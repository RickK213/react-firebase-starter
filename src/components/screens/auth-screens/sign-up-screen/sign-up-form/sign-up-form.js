import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import { withFirebase } from '../../../../firebase';
import { ROUTES } from '../../../../../constants/routes';
import { inputStyle, buttonStyle, formStyle } from '../../auth-screen-styles';

export const INITIAL_STATE = {
  username: '',
  email: '',
  passwordOne: '',
  passwordTwo: '',
  error: null
};

export class SignUpFormComponent extends Component {
  static propTypes = {
    firebase: PropTypes.object,
    history: PropTypes.object
  };

  static defaultProps = {
    firebase: null,
    history: null
  };

  constructor(props) {
    super(props);
    this.state = INITIAL_STATE;
    this.handleOnChange = this.handleOnChange.bind(this);
    this.handleOnSubmit = this.handleOnSubmit.bind(this);
  }

  handleOnChange(event) {
    if (!event) return;
    this.setState({ [event.target.name]: event.target.value });
  }

  handleOnSubmit(event) {
    const { email, passwordOne } = this.state;
    const { firebase, history } = this.props;
    firebase
      .doCreateUserWithEmailAndPassword(email, passwordOne)
      .then(() => {
        this.setState({ ...INITIAL_STATE });
        history.push(ROUTES.HOME.path);
      })
      .catch(error => {
        this.setState({ error });
        throw error;
      });
    event.preventDefault();
  }

  render() {
    const { email, error, passwordOne, passwordTwo, username } = this.state;

    const isInvalid =
      passwordOne !== passwordTwo ||
      passwordOne === '' ||
      email === '' ||
      username === '';

    return (
      <form onSubmit={this.handleOnSubmit} style={formStyle}>
        <input
          name="username"
          onChange={this.handleOnChange}
          placeholder="Username"
          style={inputStyle}
          type="text"
          value={username}
        />
        <input
          name="email"
          onChange={this.handleOnChange}
          placeholder="Email Address"
          style={inputStyle}
          type="email"
          value={email}
        />
        <input
          name="passwordOne"
          onChange={this.handleOnChange}
          placeholder="Password"
          style={inputStyle}
          type="password"
          value={passwordOne}
        />
        <input
          name="passwordTwo"
          onChange={this.handleOnChange}
          placeholder="Confirm Password"
          style={inputStyle}
          type="password"
          value={passwordTwo}
        />

        <button type="submit" disabled={isInvalid} style={buttonStyle}>
          Sign Up
        </button>

        {error && <p>{error.message}</p>}
      </form>
    );
  }
}

export const SignUpForm = withRouter(withFirebase(SignUpFormComponent));
