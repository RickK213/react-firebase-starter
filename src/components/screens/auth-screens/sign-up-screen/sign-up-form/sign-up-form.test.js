/* global React */
import ShallowRenderer from 'react-test-renderer/shallow';
import { createMemoryHistory } from 'history';
import { SignUpFormComponent } from './sign-up-form';

describe('<SignUpFormComponent />', () => {
  const renderer = new ShallowRenderer();
  const history = createMemoryHistory({ keyLength: 0 });

  it('should render correctly', () => {
    renderer.render(<SignUpFormComponent history={history} />);
    const component = renderer.getRenderOutput();
    expect(component).toMatchSnapshot();
  });
});
