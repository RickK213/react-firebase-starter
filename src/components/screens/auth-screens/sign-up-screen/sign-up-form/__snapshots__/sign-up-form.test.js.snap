// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUpFormComponent /> should render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    name="username"
    onChange={[Function]}
    placeholder="Username"
    type="text"
    value=""
  />
  <input
    name="email"
    onChange={[Function]}
    placeholder="Email Address"
    type="email"
    value=""
  />
  <input
    name="passwordOne"
    onChange={[Function]}
    placeholder="Password"
    type="password"
    value=""
  />
  <input
    name="passwordTwo"
    onChange={[Function]}
    placeholder="Confirm Password"
    type="password"
    value=""
  />
  <button
    disabled={true}
    type="submit"
  >
    Sign Up
  </button>
</form>
`;
